components/bl-cascadeSelect-component/src/subcomponent.js

```diff
@@ -0,0 +1,86 @@
+import { findParentItem } from './helpers';
+
+const { cn } = BackendlessUI.CSSUtils;
+
+export function Cascade(props) {
+  const { isOpen, itemsCascade, parentItems, levelOfNesting, openItemHandler, openCascadeHandler, selected } = props;
+
+  if (!isOpen) {
+    return null;
+  }
+
+  return (
+    <ul
+      className="cascade-select__list"
+      style={ { top: levelOfNesting ? 0 : '100%', left: levelOfNesting ? '100%' : 0 } }>
+      { itemsCascade.map(item => (
+        <CascadeItem
+          item={ item }
+          openCascadeHandler={ openCascadeHandler }
+          parentItems={ parentItems }
+          openItemHandler={ openItemHandler }
+          selected={ selected }
+        />
+      )) }
+    </ul>
+  );
+}
+
+export function CascadeItem({ item, parentItems, openItemHandler, openCascadeHandler, selected }) {
+  const { name, levelOfNesting, children, code } = item;
+
+  if (children) {
+    const { isOpen } = findParentItem(parentItems, item);
```

Comment:
would be better to not run it in runtime

Reply:
It is not very clear how to implement it, but I came up with a more optimized solution for this function